presto-password-authenticators/src/main/java/io/prestosql/plugin/password/ldap/LdapAuthenticator.java

```diff
@@ -196,7 +196,7 @@ static boolean containsSpecialCharacters(String user)
         return SPECIAL_CHARACTERS.matchesAnyOf(user);
     }
 
-    private String validateGroupMembership(String user, DirContext context)
+    private String validateGroupMembership(String user, DirContext context, boolean firstMatch)
```

Comment:
If we're going to stop on first match, we should not return anything. 
Otherwise the returned value is misleading (should not be used for anything), error-prine.

- Maybe make this method return `Stream` 
- where you currently `firstMatch`, you would just check stream is not empty
- where you currently `!firstMatch`, you'd limit 2, collect to list and throw if not singleton